app/src/androidTest/java/com/google/samples/apps/topeka/activity/SignInActivityTest.java

```diff
@@ -65,15 +71,33 @@ public void checkFab_initiallyNotDisplayed() {
     }
 
     @Test
+    public void signIn_checkIfEditTextViewsAreInitiallyEmpty() {
+        onView(withId(R.id.first_name)).check(matches(withText(isEmptyOrNullString())));
+        onView(withId(R.id.last_initial)).check(matches(withText(isEmptyOrNullString())));
+    }
+
+    @Test
     public void signIn_performSuccessful() {
         inputData();
-        onView(withId(R.id.done)).check(matches(isDisplayed()));
+        onView(withId(R.id.done)).check(matches(isDisplayed())).perform(click());
+        onView(withId(R.id.toolbar_player)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar() {
+        for (int i = 0; i < 16; i++) {
+            onData(anything())
+                    .inAdapterView(withId(R.id.avatars))
+                    .atPosition(15)
+                    .check(matches(isDisplayed()))
```

Comment:
Refactor to have only one check per test.
Also, don't use magic numbers. If you want to perform a click on the last avatar, which you always do at the moment, use [Avatar](https://github.com/googlesamples/android-topeka/blob/master/app/src/main/java/com/google/samples/apps/topeka/model/Avatar.java) as your point of reference.